projects/storefrontlib/src/cms-components/checkout/guards/checkout-auth.guard.ts

```diff
@@ -36,32 +36,41 @@ export class CheckoutAuthGuard implements CanActivate {
       this.authService.getUserToken(),
       this.activeCartService.getAssignedUser(),
       this.userService.get(),
+      this.activeCartService.isStable(),
     ]).pipe(
-      map(([token, cartUser, user]: [UserToken, User, User | B2BUser]) => {
-        if (!token.access_token) {
-          if (this.activeCartService.isGuestCart()) {
-            return Boolean(cartUser);
-          }
-          if (this.checkoutConfigService.isGuestCheckout()) {
-            this.routingService.go({ cxRoute: 'login' }, { forced: true });
-          } else {
-            this.routingService.go({ cxRoute: 'login' });
-          }
-          this.authRedirectService.reportAuthGuard();
-        } else if ('roles' in user) {
-          const roles = (<B2BUser>user).roles;
-          if (roles.includes(B2BUserGroup.B2B_CUSTOMER_GROUP)) {
-            return true;
-          } else {
-            this.globalMessageService.add(
-              { key: 'checkout.invalid.accountType' },
-              GlobalMessageType.MSG_TYPE_WARNING
-            );
-            return false;
+      filter(([, , , isStable]) => !!isStable),
```

Comment:
We prefer to use Boolean(arg) instead of !! (it's more readable)

Reply:
replaced !! with boolean().